File: javascript/examples/unary-methods.ts
import { LaserstreamClient, CommitmentLevel } from '../client';

async function main() {
  // Create once and reuse: every call shares one connection.
  const client = new LaserstreamClient({
    apiKey: process.env.HELIUS_API_KEY || 'your-api-key',
    endpoint: process.env.LASERSTREAM_ENDPOINT || 'your-endpoint',
    timeoutMs: 10_000, // per-call deadline (default 30000)
  });

  const { slot } = await client.getSlot(CommitmentLevel.CONFIRMED);
  console.log('slot (confirmed): ', slot);

  const { blockHeight } = await client.getBlockHeight();
  console.log('block height:     ', blockHeight);

  const bh = await client.getLatestBlockhash(CommitmentLevel.FINALIZED);
  console.log(`latest blockhash:  ${bh.blockhash} (slot ${bh.slot}, last valid height ${bh.lastValidBlockHeight})`);

  const { valid } = await client.isBlockhashValid(bh.blockhash);
  console.log('blockhash valid:  ', valid);

  console.log('version:          ', (await client.getVersion()).version);
  console.log('ping:             ', (await client.ping(1)).count);
  console.log('replay from slot: ', (await client.subscribeReplayInfo()).firstAvailable);

  client.close();
}

main().catch((err) => {
  console.error(err);
  process.exit(1);
});

File: rust/examples/unary_methods.rs
use helius_laserstream::grpc::CommitmentLevel;
use helius_laserstream::{LaserstreamClient, LaserstreamConfig};
use std::env;

#[tokio::main]
async fn main() -> Result<(), Box<dyn std::error::Error>> {
    dotenv::from_path("../.env").ok();

    let api_key = env::var("LASERSTREAM_PRODUCTION_API_KEY")
        .or_else(|_| env::var("HELIUS_API_KEY"))
        .expect("LASERSTREAM_PRODUCTION_API_KEY or HELIUS_API_KEY not set");
    let endpoint = env::var("LASERSTREAM_PRODUCTION_ENDPOINT")
        .or_else(|_| env::var("LASERSTREAM_ENDPOINT"))
        .expect("LASERSTREAM_PRODUCTION_ENDPOINT or LASERSTREAM_ENDPOINT not set");

    // One client, reused for every call (cheap to clone, shares the connection).
    let client = LaserstreamClient::connect(LaserstreamConfig::new(endpoint, api_key)).await?;

    let slot = client.get_slot(Some(CommitmentLevel::Confirmed)).await?;
    println!("slot (confirmed):  {}", slot.slot);

    let height = client.get_block_height(None).await?;
    println!("block height:      {}", height.block_height);

    let bh = client.get_latest_blockhash(Some(CommitmentLevel::Finalized)).await?;
    println!(
        "latest blockhash:  {} (slot {}, last valid height {})",
        bh.blockhash, bh.slot, bh.last_valid_block_height
    );

    let valid = client.is_blockhash_valid(&bh.blockhash, None).await?;
    println!("blockhash valid:   {} (checked at slot {})", valid.valid, valid.slot);

    println!("version:           {}", client.get_version().await?.version);
    println!("ping:              {}", client.ping(1).await?.count);
    println!(
        "replay from slot:  {:?}",
        client.subscribe_replay_info().await?.first_available
    );

    Ok(())
}
